packages/upset/src/components/Root.tsx

```diff
@@ -155,6 +159,12 @@ export const Root: FC<Props> = ({
           ${baseStyle};
         `}
       >
+        <Backdrop
+          sx={{ color: '#fff', zIndex: 100 }}
+          open={generating}
+        >
+          <CircularProgress color="inherit" />
+        </Backdrop>
```

Comment:
Is it worth making the whole app unusable for a second every time something changes and triggers the alt text? Perhaps it would be better to have the loading progress on the alt text instead on the whole app?

Reply:
I initially tried to put this in the alt-text sidebar only, but it was rendering over the entire view. After switching from a provenance.currentChange function to the useEffect, the speed has been drastically increased. This removes the need for a loader.